hub/src/server/utils.ts

```diff
@@ -63,3 +62,48 @@ export function timeout(milliseconds: number): Promise<void> {
     }, milliseconds)
   })
 }
+
+
+export interface AsyncMutex {
+  id: string;
+  owner: Promise<void>;
+  alive: boolean;
+}
+
+export class AsyncMutexScope {
+
+  private readonly _opened: Map<string, AsyncMutex> = new Map()
```

Comment:
Why use AsyncMutex instead of using a `Set<string>` that only stores values if the mutex is currently held?

Reply:
`AsyncMutex` is just a lightweight placeholder object with possibly useful fields (currently not used since everything is handled in the `tryAcquire` method). Also thought `Map` was more performant than `Set<T>` but looks like that is not the case. Will simplify  👍

Reply:
Addressed in https://github.com/blockstack/gaia/pull/247/commits/7899979446a0818c925c3b75e94de710c1224990